plugins/multiexternalwrench/src/applymultiexternalwrench.cpp

```diff
@@ -0,0 +1,175 @@
+        yError ( "ERROR Yarp Network was not found active in ApplyExternalWrench plugin" );
+        return;
+    }
+
+    // Copy the pointer to the model to access later from UpdateChild
+    this->m_myModel = _model;
+
+    bool configuration_loaded = false;
+
+    // Read robot name
+    if ( _sdf->HasElement ( "robotNamefromConfigFile" ) ) {
+        std::string iniRobotName      = _sdf->Get<std::string> ( "robotNamefromConfigFile" );
+        std::string iniRobotNamePath =  gazebo::common::SystemPaths::Instance()->FindFileURI ( iniRobotName );
+
+        if ( iniRobotNamePath != "" && this->m_iniParams.fromConfigFile ( iniRobotNamePath.c_str() ) ) {
+            yarp::os::Value robotNameParam = m_iniParams.find ( "gazeboYarpPluginsRobotName" );
+            this->robotName = robotNameParam.asString();
+            
+            m_rpcThread.setRobotName  ( robotName );
+            m_rpcThread.setRobotModel(_model);
+            configuration_loaded = true;
+        } else {
+            yError ( "ERROR trying to get robot configuration file" );
+            return;
+        }
+    } else {
+            this->robotName = _model->GetName();
+            m_rpcThread.setRobotName ( robotName );
+            configuration_loaded = true;
+    }
+
+    // Starting RPC thread to read desired wrench to be applied
+    if ( !m_rpcThread.start() ) {
+        yError ( "ERROR: rpcThread did not start correctly" );
+    }
+    
+    this->m_updateConnection = event::Events::ConnectWorldUpdateBegin(boost::bind(&ApplyMultiExternalWrench::applyWrenchs,this));
+}
+
+}
+
+void ApplyMultiExternalWrench::applyWrenchs()
+{
+    //Now check for duration done flag from all the wrench threads and removes the ones that are done
+    //yInfo() << "Applying external wrenches";
+    //yInfo() << "Number of external wrenches : " << m_rpcThread.wrenchThreads.size();
+    m_rpcThread.m_lock.lock();
+    for(int i = 0; i < m_rpcThread.wrenchesVectorPtr->size() ; i++)
+    {
+        bool duration_check = m_rpcThread.wrenchesVectorPtr->at(i)->duration_done;
+        if(duration_check==false)
+        {
+            m_rpcThread.wrenchesVectorPtr->at(i)->applyWrench();
+        }
+        else
+        {
+            //yInfo() << "External wrench duration done";
+        }
+    }
+    m_rpcThread.m_lock.unlock();
+}
+
+void RPCServerThread::setRobotModel(physics::ModelPtr robotModel)
+{
+    m_robotModel = robotModel;
+}
+
+
+bool RPCServerThread::threadInit()
+{
+
+    if ( !m_rpcPort.open ( std::string ( "/"+m_robotName + "/applyMultiExternalWrench/rpc:i" ).c_str() ) ) {
+        yError ( "ERROR opening RPC port /applyExternalWrench" );
+        return false;
+    }
+  
+    return true;
+}
+
+void RPCServerThread::run()
+{
+    while(!isStopping())
+    {
+        yarp::os::Bottle command;
+        m_rpcPort.read ( command,true );
+        if(command.get(0).asString() == "help")
+        {
+            this->m_reply.addVocab ( yarp::os::Vocab::encode ( "many" ) );
```

Comment:
I see. If this is the case and this plugin just adds a feature in a backward compatible way, perhaps we can just merge this plugin in the `externalwrench` one instead of having two plugins that do almost the same thing?